typescript/cli/src/tests/cosmosnative/core/core-deploy.e2e-test.ts

```diff
@@ -0,0 +1,343 @@
+        undefined,
+        undefined,
+        HYP_KEY,
+      ).stdio('pipe');
+
+      const finalOutput = await handlePrompts(output, steps);
+
+      expect(finalOutput.exitCode).to.equal(0);
+
+      const coreConfig: CoreConfig = await readCoreConfig(
+        REGISTRY_PATH,
+        CHAIN_NAME_1,
+        CORE_READ_CONFIG_PATH_1,
+      );
+      expect(coreConfig.owner).to.equal(initialOwnerAddress);
+      expect(coreConfig.proxyAdmin?.owner).to.be.undefined;
+      // Assuming that the ProtocolFeeHook is used for deployment
+      const defaultHookConfig = coreConfig.defaultHook as Exclude<
+        CoreConfig['defaultHook'],
+        string
+      >;
+      expect(defaultHookConfig.type).to.equal(
+        HookType.INTERCHAIN_GAS_PAYMASTER,
+      );
+      expect((defaultHookConfig as IgpConfig).owner).to.equal(
+        initialOwnerAddress,
+      );
+    });
+  });
+
+  describe('hyperlane cosmosnative core deploy --yes --key ...', () => {
+    it('should create a core deployment with the signer as the mailbox owner', async () => {
+      await hyperlaneCoreDeploy(
+        REGISTRY_PATH,
+        HYP_KEY,
+        CHAIN_NAME_1,
+        CORE_CONFIG_PATH,
+      );
+
+      const coreConfig: CoreConfig = await readCoreConfig(
+        REGISTRY_PATH,
+        CHAIN_NAME_1,
+        CORE_READ_CONFIG_PATH_1,
+      );
+
+      expect(coreConfig.owner).to.equal(initialOwnerAddress);
+      expect(coreConfig.proxyAdmin?.owner).to.be.undefined;
+      // Assuming that the ProtocolFeeHook is used for deployment
+      expect((coreConfig.defaultHook as IgpConfig).owner).to.equal(
+        initialOwnerAddress,
+      );
+    });
+
+    it('should create a core deployment with the mailbox owner set to the address in the config', async () => {
+      const coreConfig: CoreConfig = await readYamlOrJson(CORE_CONFIG_PATH);
+
+      const newOwner = await randomCosmosAddress(
+        chainMetadata.bech32Prefix || 'hyp',
+      );
+
+      coreConfig.owner = newOwner;
+      writeYamlOrJson(CORE_READ_CONFIG_PATH_1, coreConfig);
+
+      // Deploy the core contracts with the updated mailbox owner
+      await hyperlaneCoreDeploy(
+        REGISTRY_PATH,
+        HYP_KEY,
+        CHAIN_NAME_1,
+        CORE_READ_CONFIG_PATH_1,
+      );
+
+      // Verify that the owner has been set correctly without modifying any other owner values
+      const updatedConfig: CoreConfig = await readCoreConfig(
+        REGISTRY_PATH,
+        CHAIN_NAME_1,
+        CORE_READ_CONFIG_PATH_1,
+      );
+
+      expect(updatedConfig.owner.toLowerCase()).to.equal(newOwner);
+      expect(updatedConfig.proxyAdmin?.owner).to.be.undefined;
+      // Assuming that the ProtocolFeeHook is used for deployment
+      expect((updatedConfig.defaultHook as IgpConfig).owner).to.equal(
+        initialOwnerAddress,
+      );
+    });
+
+    it('should create a core deployment with ProxyAdmin owner of the mailbox set to the address in the config', async () => {
```

Comment:
Does the existence of proxy owner make sense in the cosmos world? i thought we didn't have one